athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -506,4 +509,39 @@ protected TableName getTableFromMetadata(final String catalogName, final TableNa
         }
         return tableName;
     }
+    @Override
+    public ListSchemasResponse doListSchemaNames(final BlockAllocator blockAllocator, final ListSchemasRequest listSchemasRequest)
+    {
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
+            LOGGER.info("{}: List schema names for Catalog {}", listSchemasRequest.getQueryId(), listSchemasRequest.getCatalogName());
+            return new ListSchemasResponse(listSchemasRequest.getCatalogName(), listDatabaseNames(connection));
+        }
+        catch (SQLException sqlException) {
+            throw new RuntimeException(sqlException.getErrorCode() + ": " + sqlException.getMessage());
+        }
+    }
+    protected static Set<String>  listDatabaseNames(final Connection jdbcConnection)
+            throws SQLException
+    {
+        try (ResultSet resultSet = jdbcConnection.getMetaData().getSchemas()) {
+            ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+            ImmutableSet.Builder<String> catalogNames = ImmutableSet.builder();
+            String inputCatalogName = jdbcConnection.getCatalog();
+            String inputSchemaName = jdbcConnection.getSchema();
+            while (resultSet.next()) {
+                String schemaName = resultSet.getString("TABLE_SCHEM");
+                String catalogName = resultSet.getString("TABLE_CATALOG");
+                // skip internal schemas
+                if (inputSchemaName != null && schemaName.equals(inputSchemaName)  && !schemaName.equals("information_schema") && catalogName.equals(inputCatalogName)) {
+                    schemaNames.add(schemaName);
+                    catalogNames.add(catalogName);
+                }
+                else if (inputSchemaName == null && !schemaName.equals("information_schema") && catalogName.equals(inputCatalogName)) {
+                    schemaNames.add(schemaName);
+                    catalogNames.add(catalogName);
+                }
```

Comment:
I think this can be simplified to:
```
boolean shouldAddSchema =
    ((inputSchemaName == null) || schemaName.equals(inputSchemaName)) && 
        (!schemaName.equals("information_schema") && catalogName.equals(inputCatalogName));

if (shouldAddSchema) {
    schemaNames.add(schemaName);
}

```

Reply:
Hi Henry, i have addressed this now.